packages/dashboard/dev-server/all.ts

```diff
@@ -0,0 +1,48 @@
+import concurrently from "concurrently";
+
+function notice(text: string) {
+  console.log(`\x1b[34m\x1b[1m${text}\x1b[0m`);
+}
+
+async function startServers() {
+  notice("Press ^C or Q to exit\nStarting servers...");
+
+  const { commands, result } = concurrently(
+    [
+      {
+        command: "yarn start:server",
+        name: "dashboard",
+        prefixColor: "bgMagenta.bold"
+      },
+      {
+        command: "yarn start:react",
+        name: "webpack",
+        prefixColor: "cyan.bgBlack.bold"
+      }
+    ],
+    {
+      killOthers: ["success", "failure"],
+      handleInput: true
+    }
+  );
+
+  const kill = () => commands.forEach(command => void command.kill("SIGINT"));
+
+  process.stdin.setRawMode(true);
+  process.stdin.on("data", async data => {
+    const isControlC = data.equals(Buffer.from("\x03"));
+    const isQ = /^q$/i.test(data.toString().trim());
+
+    if (isControlC || isQ) {
+      notice("Stopping servers...");
+      kill();
+      try {
+        await result;
+      } catch (err) {
+        /* We don't really care. */
+      }
+    }
+  });
+}
```

Comment:
It might be better to just register the `kill` function as a `SIGINT` handler via `process.on("SIGINT", kill);` That also doesn't require you to put stdin in raw mode.

https://nodejs.org/api/process.html#signal-events

Reply:
That would be cleaner but the reason for this is that I found not using raw mode / handling SIGINT directly can cause something like this:
```sh
$ yarn start
[dashboard] Dashboard server running at http://localhost:24012
[webpack] Compiling: 3%
[webpack] ...
^C  # Send kill signal
[dashboard] yarn start:server exited with code SIGINT                                                                                                                                      
--> Sending SIGTERM to other processes..
$ 
# Some time later
[webpack] Finished compiling
[webpack] yarn start:react exited with code SIGINT
```

Reply:
Ah fair enough. Just wanted to make sure that you'd considered it.